### PR TITLE
THEMES-1333 calling replace-token-reference from root-tokens

### PR DESCRIPTION
**Pull request by malavikakoppula**
## Ticket

- [THEMES-1333](https://arcpublishing.atlassian.net/browse/THEMES-1333)

## Description
Currently we are handling this issue in the stylebuilder script, but it will be preferred to handle it in the SASS library in arc-themes-components. This will make it easier to maintain our replacement logic in the future.

## Acceptance Criteria

- The replace-token-reference mixin in the arc-themes-components SASS library should also replace uses of $global tokens in the alias definitions.

- The resolveSassTokens function should be removed from the stylebuilder script to keep this functionality in one place.

## Test Steps

1. Checkout branch - `git checkout THEMES-1333-sass-global`
2. link feature-pack repo with fusion/cli
3. start fusion with any block to check the global tokens getting applied.

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [ ] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed relevant documentation has been updated/added.
- [ ] Add label - **ready for review** when the pull request is ready for someone to begin reviewing

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] All GitHub Actions pass
- [ ] Ran the code locally based on the test instructions.
- [ ] Checked Chromatic for Storybook changes, accepted the updates if acceptable
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.
- [ ] Approve and Add label - **ready to merge** if you are happy with the pull request
- [ ] Want another reviewer? Add the label **additional review**

[THEMES-1333]: https://arcpublishing.atlassian.net/browse/THEMES-1333?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by vgalatro** [comment]
@malavikakoppula , I'm having trouble getting this to run at all with both the branches checked out. There's errors in the SCSS that prevents the styles from fully building. Specifically, I'm seeing webpack error out on `$global.neutral-7` in the styles for the ad block. If you have some time could you meet with me to show me how this should be working? Thanks.

**Review by nschubach — APPROVED** [review]

**Comment by malavikakoppula** [comment]
> @malavikakoppula , I'm having trouble getting this to run at all with both the branches checked out. There's errors in the SCSS that prevents the styles from fully building. Specifically, I'm seeing webpack error out on `$global.neutral-7` in the styles for the ad block. If you have some time could you meet with me to show me how this should be working? Thanks.

@vgalatro Fixed this `$global.neutral-7` error in both components and stylebuilder repos

File: src/scss/root/_index.scss [hunk 1]
```diff
@@ -164,6 +164,15 @@ $component-map: get-all-component-definitions($tokens, $breakpoints);
...
 						--#{$prefix}-#{$componentName}-#{$cProperty}: #{maintain-quotes($new-value)};
+					} @else {
+						@if type-of($cValue) == map {
+							@each $property, $value in $cValue {
+								$new-value: replace-token-reference($cValue);
+								--#{$prefix}-#{$componentName}-#{$cProperty}-#{$property}: #{maintain-quotes(
+										$new-value
+									)};
+							}
+						}
```
Comment (vgalatro) [inline src/scss/root/_index.scss]: @malavikakoppula , I've been trying to figure out why this `@else` condition is here. As far as I can tell it never even gets called. Was there an issue this solved?

Comment (malavikakoppula) [reply src/scss/root/_index.scss]: @vgalatro Forgot to delete that earlier. It's removed now [58ff12b](https://github.com/WPMedia/arc-themes-components/pull/209/commits/58ff12be97fe53f95eda50fcb287635f9f52af39)

File: src/scss/index.test.scss [hunk 2]
```diff
@@ -97,27 +97,6 @@ $tokens: (
...
-			}
-			@include expect {
-				:root {
-					--color: #f00;
-					--font-family: "Inter";
-				}
-			}
-		}
-	}
-
```
Comment (vgalatro) [inline src/scss/index.test.scss]: @malavikakoppula , we should be keeping this test, was there a reason it was removed? I added it back in and it does pass just fine.

**Review by vgalatro — CHANGES_REQUESTED** [review]
Just need that test added back in, then this is good to merge!

File: src/scss/index.test.scss (hunk 2 above)
Comment (malavikakoppula) [reply src/scss/index.test.scss]: @vgalatro Added it back [acbd7ef](https://github.com/WPMedia/arc-themes-components/pull/209/commits/acbd7efb41754a126fd656edfe4897262e27b231)

**Review by vgalatro — APPROVED** [review]
Everything looks good now!

**Pull request merged**